.github/workflows/formatting.yml

```diff
@@ -0,0 +1,15 @@
+name: Code formatting
+on: push
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '18.x'
+      - name: Install dependencies
+        run: npm ci
```

Comment:
**thought:** At some point we may need to move tasks around between actions to cut down on the number of times we're running `npm ci` in CI

Reply:
That's a great thought.

The `setup-node` action supports caching. Added in ef95346c.

**Monorepo dependency cache**
Monorepo dependencies can be cached too. Example below.

<details><summary>Details</summary>
<p></p>

```yml
steps:
- uses: actions/setup-node@v4
  with:
    node-version: '18.x'
    cache: 'npm'
    cache-dependency-path: '**/package-lock.json'
- run: npm ci
- run: npm test
```
**Source:** https://github.com/actions/setup-node?tab=readme-ov-file#caching-global-packages-data
</details>